Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Collections/GridObjectCollection.cs

```diff
@@ -444,7 +476,7 @@ public void OnValidate()
                 }
 
                 // If we have an old asset, then rows could be != default value, columns would be default value
-                bool upgradeScenarioA = rows != defaultValueRowsCols && columns == defaultValueRowsCols 
+                bool upgradeScenarioA = rows != defaultValueRowsCols && columns == defaultValueRowsCols
```

Comment:
I'm probably missing something here but if I understand this correctly you're determining if we have to upgrade this asset purely by checking if those values are set to the wrong defaults, which could be problematic in future. After version 2.2 there might be grid object collections that have those values set up on purpose by the user and then we're upgrading those assets and changing the setup the user wanted to have on every scene load.

Reply:
Added asset versions